### PR TITLE
Reward center merge

### PR DESCRIPTION
**Pull request by gchatz22**

### CODE REVIEW

File: components/StakePoolInfo.tsx [hunk 1]
```diff
@@ -154,8 +157,9 @@ export const StakePoolInfo = () => {
                       '???'}
                   </div>
                   <div className="text-xs text-gray-500">
-                    {rewardDistributorData.data.parsed.kind ===
-                    RewardDistributorKind.Mint
+                    {rewardDistributorData.data.parsed?.kind ===
+                      RewardDistributorKind.Mint &&
+                    !isRewardDistributorV2(rewardDistributorData.data.parsed)
```
Comment (jpbogle) [inline components/StakePoolInfo.tsx:162]: why not v2 reward distributor here?

File: handlers/useHandleClaimRewards.ts [hunk 2]
```diff
@@ -55,14 +61,36 @@ export const useHandleClaimRewards = () => {
...
+                    {
+                      mintId: token.stakeEntry.parsed!.stakeMint,
+                    },
+                  ],
+                  rewardDistributorData.data
+                    ? [rewardDistributorData.data.pubkey]
+                    : []
+                )
+              } else {
+                console.log('heyyyyy')
```
Comment (jpbogle) [inline handlers/useHandleClaimRewards.ts:81]:
```suggestion
```

File: handlers/useHandleClaimRewards.ts [hunk 3]
```diff
@@ -55,14 +61,36 @@ export const useHandleClaimRewards = () => {
...
+                claimTxs = await claimRewardsV2(
+                  connection,
+                  wallet,
+                  stakePool.parsed!.identifier,
+                  [
+                    {
+                      mintId: token.stakeEntry.parsed!.stakeMint,
+                    },
+                  ],
+                  rewardDistributorData.data
```
Comment (jpbogle) [inline handlers/useHandleClaimRewards.ts:76]: souldnt need this cuz its checked above maybe save as a variable

File: handlers/useHandleUnstake.ts [hunk 4]
```diff
@@ -66,11 +67,27 @@ export const useHandleUnstake = (callback?: () => void) => {
...
+              if (!token.stakeEntry.parsed?.stakeMint)
+                throw 'No stake mint found for stake entry'
+              if (isStakePoolV2(stakePool.parsed!)) {
+                if (!stakePool.parsed) throw 'No stake pool parsed data'
+                const txs = await unstakeV2(
+                  connection,
+                  wallet,
+                  stakePool.parsed?.identifier,
+                  [{ mintId: token.stakeEntry.parsed?.stakeMint }]
+                ) // TODO Handle fungible
```
Comment (jpbogle) [inline handlers/useHandleUnstake.ts:80]: hmm

File: hooks/useAllStakeEntries.tsx [hunk 5]
```diff
@@ -9,10 +12,31 @@ import { useStakePoolId } from './useStakePoolId'
...
+
+  return useQuery<
+    Pick<IdlAccountData<'stakeEntry'>, 'pubkey' | 'parsed'>[] | undefined
+  >(['useAllStakeEntries', stakePoolId?.toString()], async () => {
+    const stakeEntriesV1 = await getAllStakeEntries(secondaryConnection)
+    const program = rewardsCenterProgram(
+      secondaryConnection,
+      asEmptyAnchorWallet(wallet)
+    )
+    const stakeEntriesV2 = await program.account.stakeEntry.all()
```
Comment (jpbogle) [inline hooks/useAllStakeEntries.tsx:25]: woah lol

Comment (jpbogle) [reply hooks/useAllStakeEntries.tsx:25]: cool

File: common/Wallets.tsx [hunk 6]
```diff
@@ -8,3 +10,14 @@ export const asWallet = (wallet: WalletContextState): Wallet => {
...
+export const asEmptyAnchorWallet = (
+  wallet: WalletContextState
+): AnchorWallet => {
+  return {
+    signTransaction: wallet.signTransaction!,
+    signAllTransactions: wallet.signAllTransactions!,
+    publicKey: wallet.publicKey!,
+    payer: new Keypair(),
+  }
+}
```
Comment (jpbogle) [inline common/Wallets.tsx:23]: shuldnt be needed if we use the Wallet interface

File: handlers/useHandleUnstake.ts (hunk 4 above)
Comment (gchatz22) [reply handlers/useHandleUnstake.ts:80]: wdym

File: components/StakePoolInfo.tsx (hunk 1 above)
Comment (gchatz22) [reply components/StakePoolInfo.tsx:162]: confused on what you mean here

Comment (gchatz22) [reply components/StakePoolInfo.tsx:162]: but basically saying if not V2 pool and is of type `Mint` use the mint supply else in any other case use the token account balance

File: handlers/useHandleClaimRewards.ts (hunk 3 above)
Comment (gchatz22) [reply handlers/useHandleClaimRewards.ts:76]: yah haha agreed, its one of those scenarios TS doesnt recognize the condition was checked above

File: common/Wallets.tsx (hunk 6 above)
Comment (gchatz22) [reply common/Wallets.tsx:23]: yah i mean i know its just that i couldnt get it to work with the Wallet interface being imported from `cjs` i know what you are talking about. would be chill if youd wanna look at this together, could just be something small and dumb end of the day

**Pull request merged**